addons/im_livechat/controllers/main.py

```diff
@@ -219,13 +220,14 @@ def feedback(self, uuid, rate, reason=None, **kwargs):
                 'feedback': reason,
                 'is_internal': False,
             }
-            if not channel.rating_ids:
+            # sudo: rating.rating - visitor can access rating to give its feedback
```

Comment:
this line in particular is not to "give feedback" but to check if it's already given (which is also ok)